src/string_object.cpp

```diff
@@ -32,10 +32,10 @@ char *StringObject::next_char(Env *env, char *buffer, size_t *index) {
         buffer[0] = str[i];
         if (((unsigned char)buffer[0] >> 3) == 30) { // 11110xxx, 4 bytes
             if (i + 3 >= len) raise_encoding_invalid_byte_sequence_error(env, i);
-            buffer[1] = str[++i];
-            buffer[2] = str[++i];
-            buffer[3] = str[++i];
-            buffer[4] = 0;
+            for (size_t j = 1; j < 5; j++)
+            {
+                buffer[j] = str[j];
+            }
```

Comment:
The point of next_char() is to (1) fill the `buffer` with the bytes of the character, and (2) increment the `index` (pointer to) integer to the next byte position. Your change here makes it so that no longer happens.

I'm sure there are bugs here -- we haven't exposed this code to specs yet. But the code change here doesn't seem right to me.